core/src/main/java/io/substrait/relation/VirtualTableScan.java

```diff
@@ -36,11 +33,6 @@ protected void check() {
             .allMatch(r -> r.getType().accept(new NamedFieldCountingTypeVisitor()) == names.size());
   }
 
-  @Override
-  public final NamedStruct getInitialSchema() {
-    return NamedStruct.of(getDfsNames(), (Type.Struct) getRows().get(0).getType());
```

Comment:
The only benefit of this I can think of is to kind of ensure the schema matches between the rows and the initialSchema. However, given substrait allows those to be specified separately, I think it makes sense to keep that separation here?

Also, given substrait literals specify nullability on each literal, the schema might be different for row 0 than for row 1. For the producers, it's often easier to ensure the nullability is correct for the initialSchema than for each literal, since the literals are usually converted independently.

Reply:
> given substrait allows those to be specified separately, I think it makes sense to keep that separation here?

That makes sense to me and it also avoids the issue you highligh

> the schema might be different for row 0 than for row 1.